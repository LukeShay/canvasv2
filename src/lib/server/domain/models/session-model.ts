/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable global-require */
import { Model, JSONSchema, RelationMappings } from 'objection';
import Constants from '../constants';

export class SessionModel extends Model {
  static tableName = Constants.SESSIONS_TABLE;

  static idColumn = 'id';

  static get jsonSchema(): JSONSchema {
    return {
      properties: {
        device: { type: 'string' },
        expires: { type: 'date' },
        userId: { type: 'string' },
      },
      required: ['device', 'userId', 'expires'],
      type: 'object',
    };
  }

  static get relationMappings(): RelationMappings {
    // ts-expect-error
    const { UserModel } = require('./user-model');

    return {
      user: {
        join: {
          from: `${SessionModel.tableName}.userId`,
          to: `${UserModel.tableName}.${UserModel.idColumn}`,
        },
        modelClass: UserModel,
        relation: Model.BelongsToOneRelation,
      },
    };
  }
}
