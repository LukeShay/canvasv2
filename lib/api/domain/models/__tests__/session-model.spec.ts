import { SessionModel } from '../session-model';

describe('SessionModel', () => {
  it("should have idColum 'id'", () => {
    expect(SessionModel.idColumn).toEqual('id');
  });

  it('should should have jsonSchema that matches snapshot', () => {
    expect(SessionModel.jsonSchema).toMatchSnapshot();
  });

  it('should should have relationMappings that matches snapshot', () => {
    expect(SessionModel.relationMappings).toMatchSnapshot();
  });
});
