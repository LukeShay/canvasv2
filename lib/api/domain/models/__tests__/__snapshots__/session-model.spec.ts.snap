// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionModel should should have jsonSchema that matches snapshot 1`] = `
Object {
  "properties": Object {
    "device": Object {
      "type": "string",
    },
    "expires": Object {
      "type": "date",
    },
    "userId": Object {
      "type": "string",
    },
  },
  "required": Array [
    "device",
    "userId",
    "expires",
  ],
  "type": "object",
}
`;

exports[`SessionModel should should have relationMappings that matches snapshot 1`] = `
Object {
  "user": Object {
    "join": Object {
      "from": "Sessions.userId",
      "to": "Users.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
}
`;
